h/services/search_index/_queue.py

```diff
@@ -227,7 +254,4 @@ def _get_annotations_from_es(self, annotation_ids):
     @staticmethod
     def _datetime_at(delta_seconds):
         """Return the datetime at delta_seconds seconds from now."""
-        if not delta_seconds:
-            return None
-
-        return datetime.utcnow() + timedelta(seconds=delta_seconds)
+        return datetime.utcnow() + timedelta(seconds=delta_seconds or 0)
```

Comment:
It turned out there was a bug in this: if `delta_seconds` was `None` it would return `None` and then the calling function would try to put `NULL` into the non-nullable `scheduled_at` column. It should be fine to pass `None` for `scheduled_at` because it has a `server_default` but -- I think because we're using the expression language not the ORM -- SQLAlchemy explicitly tries to set the column to `NULL` which fails. Fixed it to always return a `datetime`